pkg/fleet/fleet_test.go

```diff
@@ -0,0 +1,75 @@
+package fleet_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/rancher/rancher/pkg/fleet"
+)
+
+func TestGetClusterHost(t *testing.T) {
+	testCases := []struct {
+		name         string
+		config       clientcmdapi.Config
+		expectedHost string
+		expectErr    bool
+	}{
+		// the in-cluster config case is not tested, as it would require a container with environment variables,
+		// token and cert files.
+		{
+			name: "returns host of raw config cluster with current context",
+			config: clientcmdapi.Config{
+				CurrentContext: "foo",
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"foo": &clientcmdapi.Cluster{
+						Server: "bar",
+					},
+				},
+			},
+			expectedHost: "bar",
+			expectErr:    false,
+		},
+		{
+			name: "returns host of first found configured cluster when none found with current context",
```

Comment:
This test will be flakey, because Go randomises ranging over maps...

https://go.dev/ref/spec#For_statements

Reply:
Oh indeed, nice catch, thank you!
I've amended the tests to support a slice of expected hosts.
If this LGTY, I'll propagate it to #44612.